pages/dashboard.vue

```diff
@@ -33,42 +32,59 @@
   </v-container>
 </template>
 
-<script>
+<script lang="ts">
+import Vue from 'vue'
 import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
-// local
 import UserProduct from '@/components/UserProduct.vue'
 import { ProductCode, ProductStatus } from '@/enums'
-import { getAccountProducts, getKeycloakRoles, getProductInfo } from '@/utils'
+import { fetchAccountProducts, getKeycloakRoles, getProductInfo, sleep } from '@/utils'
 
-export default {
+export default Vue.extend ({
   components: {
-    UserProduct
+    UserProduct,
   },
-  // Called every time before loading page components.
-  // TODO: need to start Token Service to refresh KC token periodically?
-  asyncData(context) {
-    if (!sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)) {
-      context.redirect('/signin')
+  asyncData ({ $config, redirect }) {
+    // if user is not logged in, redirect to home page
+    const token = sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)
+    if (!token) {
+      return redirect($config.baseURL)
     }
+
+    // if user is staff, redirect to Business Registry staff dashboard
+    try {
+      if (getKeycloakRoles().includes('staff')) {
+        return redirect($config.authURL + 'staff/dashboard/active')
+      }
+    } catch {} // ignore errors
   },
-  data() {
+  data () {
     return {
-      getProductInfo,
-      roles: getKeycloakRoles(),
-      isStaffSbc: false,
+      getProductInfo, // for use in template
       subscribedProducts: [],
     }
   },
-  async mounted() {
-    if (this.roles.includes('staff')) {
-      const redirectURL = this.$config.authURL + 'staff/dashboard/active'
-      window.location.href = redirectURL
+  computed: {
+    isLoggedIn (): boolean {
+      const token = sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)
+      return !!token
+    },
+  },
+  async mounted () {
+    // get roles
+    let roles: Array<string>
+    try {
+      roles = getKeycloakRoles()
+    } catch (e) {
+      roles = []
     }
+
+    // safety check - sometimes this page is rendered
+    // even if we redirected in asyncData()
+    if (roles.includes('staff')) return
+
+    // get products / services
     let products = []
-    // title logic
-    if (this.roles.includes('gov_account_user')) {
-      this.isStaffSbc = true
-      // get products / services
+    if (roles.includes('gov_account_user')) {
```

Comment:
this is logic that should only be in place for sbc staff not all gov account users. This will need to use the new sbc logic I mentioned above too

Reply:
Unless we are going to put that in right away?

Reply:
So, the current check for 'gov_account_user' needs the addition of the auth call you mentioned above? Is that the only change?

This logic affects the products/services tiles that are displayed on the dashboard (and potentially the header if I revert that).

Reply:
Well, I could put that in as part of this ticket, once I understand the requirements. As of right now, I've heard conflicting things.

Reply:
Yeah so it should follow what we do in ppr: 
if gov_account_user:  (this if is only here so that we don't do the call for every user)
  make auth call and check if sbc staff
  set global sbc staff to ^ and do remaining sbc staff logic off that

if they aren't sbc staff then nothing changes

Reply:
currently everywhere you see `if gov_account_user` it is just a temp for `if is_sbc_staff`

Reply:
What are you calling SBC Staff? You said there were 3 types of staff; what are they?

Reply:
bcol / registry / sbc staff. In this one bcol / registry staff are treated the same. You can call sbc staff whatever you like, but in ppr we added this to our store:
```
authorization: {
    authRoles: [],
    isSbc: false
  },
```
and then set isSbc after making the auth call in the beginning and referred to that for our logic

Reply:
I just realized what you meant haha -- bcol staff have 'helpdesk' as a role, registry staff have 'ppr_staff' as role (although they also both have the 'staff' role), sbc staff have the 'gov_account_user' role and also have Service BC in their branch name (you need to make the auth call I showed in the link above to get this)